src/components/communities/CreateCommunityPage.tsx

```diff
@@ -85,105 +68,94 @@ export default function CreateCommunityPage({}: StackNavigationProps<
   const communityDispatch = useCommunityDispatch();
   const [tokenState, tokenDispatch] = useToken();
   const accountState = useAccountState();
+  const snackbarDispatch = useSnackbarDispatch();
 
-  const submitCommunity: SubmitHandler<FormValues> = (data) => {
-    console.log({
-      ...BLANK_COMMUNITY,
-      name: data.name,
-      codeofconduct: data.codeOfConduct,
-      Admins: [accountState.account.email],
-      Members: [accountState.account.email],
-      token: tokenState.refreshToken,
+  const submitCommunity: SubmitHandler<FormValues> = async (data) => {
+    await createCommunity(
+      communityDispatch,
+      tokenDispatch,
+      tokenState.refreshToken,
+      {
+        ...BLANK_COMMUNITY,
+        name: data.name,
+        codeofconduct: data.codeOfConduct,
+        Admins: [accountState.account.email],
+        Members: [accountState.account.email],
+      }
+    );
+    snackbarDispatch({
+      type: "push",
+      state: {
+        snackbarVisible: true,
+        snackbarMessage: "Successfully Created Community",
+        queue: [],
+        snackbarType: "success",
+      },
+      dispatch: snackbarDispatch,
     });

```

Comment:
this looks good, however I would get the result returned from `createCommunity` and use that to display the snackbar message because there is a chance the request could fail.

Reply:
I don't believe that createCommunity returns a value, it just returns a Promise of type void. Not sure how we should capture the result

Reply:
we can update the promise to return true or false

Reply:
I was looking at startMatching since you do something similar to show the snackbar there, however, the implementation of startMatching is very different from createCommunity. Not quite sure how to update createCommunity to get the return status of the request

Reply:
I can add that. Just give me a few min to pull and make the change